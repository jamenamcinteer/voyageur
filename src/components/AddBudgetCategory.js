import React from "react";
import TripHeader from "./Navigation/TripBudgetHeader";
import BudgetCategoryForm from "./Forms/BudgetCategoryForm";
import { connect } from "react-redux";

const AddBudgetCategory = props => {
  const trip = props.trips.find(trip => trip._id === props.match.params.id);

  return (
    <React.Fragment>
      {trip && (
        <React.Fragment>
          <TripHeader
            title="Add Budget Category"
            theme={props.theme}
            backTo={`/trip-budget/${props.match.params.id}`}
            trip={trip}
            budgetItems={props.budgetItems}
            expenses={props.expenses}
            auth={props.auth}
          />
          <BudgetCategoryForm
            theme={props.theme}
            trip={trip}
            history={props.history}
          />
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

const mapStateToProps = (state, props) => {
  return {
    trips: state.trips,
    budgetItems: state.budgetItems,
    expenses: state.expenses,
    auth: state.auth
  };
};

export default connect(mapStateToProps)(AddBudgetCategory);
