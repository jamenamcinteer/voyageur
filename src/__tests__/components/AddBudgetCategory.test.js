import React from "react";
import { render, cleanup } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import "react-dates/initialize";
import AddBudgetCategory from "../../components/AddBudgetCategory";
import {
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
} from "../../testUtils/fixtures/mockStoreData";

function renderWithReduxAndRouter(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    ),
    store,
    history
  };
}

afterEach(cleanup);

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

test("can render with redux with custom store", () => {
  const store = createStore(() => ({
    trips,
    budgetCategories,
    budgetItems,
    expenses,
    auth
  }));
  const { asFragment } = renderWithReduxAndRouter(
    <AddBudgetCategory match={{ params: { id: "1" } }} />,
    {
      store
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
