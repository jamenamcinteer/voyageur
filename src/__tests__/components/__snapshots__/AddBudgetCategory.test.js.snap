// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          <div
            class="sc-bdVaJa hZqfbl"
          >
            <a
              href="/trip/1"
            >
              <svg
                aria-label="Go Back"
                style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
                version="1.1"
                viewBox="0 0 199.404 199.404"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
                  />
                </g>
              </svg>
            </a>
            <h1
              class="sc-bwzfXH OVfcq"
            >
              Add Budget Category
            </h1>
            <button
              class="sc-htpNat fqCEYg"
            >
              <img
                alt="Account Options"
                class="sc-bxivhb jwtarI"
                src="https://picsum.photos/id/947/45/45"
              />
            </button>
          </div>
          <div
            style="position: relative;"
          >
            <div
              style="background-image: url(https://images.unsplash.com/photo-1549317487-92cec5e76645?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjc5ODc0fQ&w=300&h=100&fit=crop&crop=focalpoint); background-repeat: no-repeat; background-size: cover; filter: brightness(0.6); height: 150px;"
            />
            <div
              style="height: 150px; padding: 0px 20px; color: rgb(255, 255, 255); font-size: 1em; font-weight: normal; position: absolute; width: 100%; top: 0px;"
            >
              <div
                style="display: flex; justify-content: space-between; align-items: center; height: 90px;"
              >
                <h2>
                  Hawaii
                </h2>
                <h3>
                  Apr 26, 1970 - Oct 1, 8307
                </h3>
              </div>
              <div
                style="width: 100%; height: 25px; position: relative; background: rgb(220, 224, 227); border-radius: 5px;"
              >
                <span
                  style="display: block; height: 100%; border-radius: 5px; position: relative; overflow: hidden; width: 29.731%; background-color: rgb(12, 124, 136);"
                />
                <div
                  style="position: absolute; top: 0px; color: white; left: 10px; line-height: 25px;"
                >
                  $595 of $2000
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-kAzzGY jucvHf"
      >
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="1"
          >
            Name of Budget Category
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="1"
            placeholder="Example: Airfare, Food, Activities, etc."
            type="text"
            value=""
          />
        </div>
        <div />
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="2"
          >
            Notes (optional)
          </label>
          <textarea
            class="sc-gZMcBi kEtTwg"
            id="2"
            type="text"
          />
        </div>
        <div
          class="sc-kGXeez cgoskb"
        >
          <div
            class="sc-kgoBCf fWbtNX"
          >
            <a
              backgroundcolor="transparent"
              border="0"
              buttontype="link"
              buttonwidth="auto"
              class="sc-fjdhpX cPSYhm"
              href="/trip/1"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-jzJRlG cphvgw"
              >
                Cancel
              </span>
            </a>
            <button
              class="sc-VigVT fLcMOh"
            >
              <span
                class="sc-jTzLTM cntDpR"
                color="#fff"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
